### PR TITLE
Added one-point and multi-point crossover functions

### PR DESCRIPTION
**Pull request by JacquesSmulders**

### CODE REVIEW

**Comment by JacquesSmulders** [comment]
monkaS

File: genetic_algorithm.py [hunk 1]
```diff
@@ -249,15 +252,60 @@ def crossover(self, chromosome1, chromosome2):
     """
 
     def one_point_crossover(self, gene1, gene2):
-        print("todo")
+
+        # Function Call Print
+        print("*********************\n* One Point Crossover *\n*********************\n")
+        children = np.arange(2 * gene1.size, dtype=object)
+        rand = random.randint(0, gene1.size)
```
Comment (SirJosh) [inline genetic_algorithm.py]: Check Codacy error here: "Standard pseudo-random generators are not suitable for security/cryptographic purposes."

File: genetic_algorithm.py [hunk 2]
```diff
@@ -249,15 +252,60 @@ def crossover(self, chromosome1, chromosome2):
...
     """
 
-    def multipoint_crossover(self, gene1, gene2):
-        print("TODO")
+    def multi_point_crossover(self, gene1, gene2):
+
+        # Function Call Print
+        print("*********************\n* Multi Point Crossover *\n*********************\n")
+        children = np.arange(2 * gene1.size, dtype=object)
+        numPoints = int(math.floor(random.uniform(self.multipoint_crossover_percentages[0], self.multipoint_crossover_percentages[1]) * gene1.size)) # choose random number of crossover points
```
Comment (SirJosh) [inline genetic_algorithm.py]: Check Codacy Error: "Standard pseudo-random generators are not suitable for security/cryptographic purposes."

File: genetic_algorithm.py [hunk 3]
```diff
@@ -249,15 +252,60 @@ def crossover(self, chromosome1, chromosome2):
...
+
+        points = np.arange(numPoints + 2, dtype=object) # store random points in array
+        points[0] = 0
+        for x in range(1, numPoints + 1):
+            points[x] = random.randint(0, gene1.size)
```
Comment (SirJosh) [inline genetic_algorithm.py]: Check Codacy Error: "Standard pseudo-random generators are not suitable for security/cryptographic purposes."

**Pull request merged**